### PR TITLE
Update workflows to use reusable-workflows

### PR DESCRIPTION
**Pull request by jongwooo**
Signed-off-by: jongwooo <jongwooo.han@gmail.com>

## Description
In the scope of this PR we update workflows in the repo to use reusable-workflows from the [repo](https://github.com/actions/reusable-workflows).

### Related Issues / PRs
[actions/setup-node#647](https://github.com/actions/setup-node/pull/647)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (add or update README or docs)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by vsvipul** [comment]
@jongwooo Thank you for opening this. Please fixed the licensed check as well by running `licensed cache` on your local setup.

**Comment by jongwooo** [comment]
@vsvipul Thank you for your review! I pushed a commit.

File: .github/workflows/check-dist.yml [hunk 1]
```diff
@@ -18,33 +12,6 @@ on:
...
-
-      # If index.js was different than expected, upload the expected version as an artifact
-      - uses: actions/upload-artifact@v3
-        if: ${{ failure() && steps.diff.conclusion == 'failure' }}
-        with:
-          name: dist
-          path: dist/
+  call-check-dist:
+    name: Check dist/
+    uses: actions/reusable-workflows/.github/workflows/check-dist.yml@main
```
[the remark this answers is not in the document]
Comment (jongwooo) [reply .github/workflows/check-dist.yml:17]: @bishal-pdMSFT Please see [actions/reusable-workflows](https://github.com/actions/reusable-workflows)

File: .github/workflows/licensed.yml [hunk 2]
```diff
@@ -7,18 +7,9 @@ on:
   pull_request:
     branches:
       - main
+  workflow_dispatch:
```
Comment (tiwarishub) [inline .github/workflows/licensed.yml:10]: Why we are adding a dispatch event? I saw in the attached setup node PR as well

**Review by tiwarishub — APPROVED** [review]
LGTM, Left one minor comment

File: .github/workflows/licensed.yml (hunk 2 above)
Comment (jongwooo) [reply .github/workflows/licensed.yml:10]: @tiwarishub The `workflow_dispatch` event allows us to manually trigger a workflow and test it before creating a PR. I added by referring to `actions/setup-node`.

**Pull request merged**